--- a/udev-md-raid-assembly.rules
+++ b/udev-md-raid-assembly.rules
@@ -25,6 +25,9 @@ GOTO="md_inc_end"

 LABEL="md_inc"

+# Disable incremental assembly to fix Debian bug #784070
+GOTO="md_inc_end"
+
 # remember you can limit what gets auto/incrementally assembled by
 # mdadm.conf(5)'s 'AUTO' and selectively whitelist using 'ARRAY'
 ACTION=="add|change", IMPORT{program}="BINDIR/mdadm --incremental --export $tempnode --offroot ${DEVLINKS}"
