Subject: Fix '--grow --continue' when using backup space on a 'spare' device
Origin: upstream, http://git.neil.brown.name/?p=mdadm.git;a=commit;h=8e7ddc5f50af00e569ef115e25c635e2d74e90f0
Bug-Debian: https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=840743
Author: Neil Brown <neilb@suse.de>

--- mdadm-3.3.2.orig/Grow.c
+++ mdadm-3.3.2/Grow.c
@@ -853,7 +853,8 @@ int reshape_prepare_fdlist(char *devname
 	for (sd = sra->devs; sd; sd = sd->next) {
 		if (sd->disk.state & (1<<MD_DISK_FAULTY))
 			continue;
-		if (sd->disk.state & (1<<MD_DISK_SYNC)) {
+		if (sd->disk.state & (1<<MD_DISK_SYNC) &&
+		    sd->disk.raid_disk < raid_disks) {
 			char *dn = map_dev(sd->disk.major,
 					   sd->disk.minor, 1);
 			fdlist[sd->disk.raid_disk]
@@ -3216,7 +3217,7 @@ started:
 	d = reshape_prepare_fdlist(devname, sra, odisks,
 				   nrdisks, blocks, backup_file,
 				   fdlist, offsets);
-	if (d < 0) {
+	if (d < odisks) {
 		goto release;
 	}
 	if ((st->ss->manage_reshape == NULL) ||
@@ -3228,7 +3229,7 @@ started:
 				       devname);
 				pr_err(" Please provide one with \"--backup=...\"\n");
 				goto release;
-			} else if (sra->array.spare_disks == 0) {
+			} else if (d == odisks) {
 				pr_err("%s: Cannot grow - "
 					"need a spare or backup-file to backup "
 					"critical section\n", devname);
